ui/src/shared/components/__tests__/object-editor.test.tsx

```diff
@@ -0,0 +1,71 @@
+import {fireEvent, render} from '@testing-library/react';
+import React, {forwardRef, useImperativeHandle} from 'react';
+
+import {ObjectEditor} from '../object-editor';
+
+// Mock the heavy Monaco editor
+jest.mock('../suspense-monaco-editor', () => {
+    return {
+        // eslint-disable-next-line react/display-name
+        SuspenseMonacoEditor: forwardRef((props: any, ref: any) => {
+            const {defaultValue, ...forward} = props;
+
+            useImperativeHandle(ref, () => ({
+                editor: {
+                    getValue: () => defaultValue,
+                    setValue: () => {},
+                    revealLineInCenter: () => {},
+                    setPosition: () => {},
+                    focus: () => {}
+                }
+            }));
+
+            /* Render a textarea so fireEvent.change works with a native `value`
+         property.  Spread `forward` *first* so we can wrap its onChange
+         and pass only the updated string back to parent code. */
+            return <textarea data-testid='editor' defaultValue={defaultValue} {...forward} onChange={e => forward.onChange?.(e.target.value)} />;
+        })
+    };
+});
```

Comment:
I'm not sure why `useImperativeHandle()` was added, but it doesn't seem to be used by the tests. Also, passing through the props isn't necessary. I tried replacing this with the following and the tests all passed:

```suggestion
jest.mock('./suspense-monaco-editor', () => ({
    // eslint-disable-next-line react/display-name, @typescript-eslint/no-unused-vars
    SuspenseMonacoEditor: forwardRef((props: any, _) => <textarea data-testid='editor' onChange={e => props.onChange(e.target.value)} />)
}));
```